src/token/erc1155/erc1155.cairo

```diff
@@ -489,44 +487,62 @@ mod ERC1155Component {
             assert(from.is_non_zero(), Errors::INVALID_SENDER);
             self.update(from, Zeroable::zero(), token_ids, values);
         }
+
+        /// Sets a new URI for all token types, by relying on the token type ID
+        /// substitution mechanism defined in the ERC1155 standard.
+        /// See https://eips.ethereum.org/EIPS/eip-1155#metadata.
+        ///
+        /// By this mechanism, any occurrence of the `\{id\}` substring in either the
+        /// URI or any of the values in the JSON file at said URI will be replaced by
+        /// clients with the token type ID.
+        ///
+        /// For example, the `https://token-cdn-domain/\{id\}.json` URI would be
+        /// interpreted by clients as
+        /// `https://token-cdn-domain/000000000000000000000000000000000000000000000000000000000004cce0.json`
+        /// for token type ID 0x4cce0.
+        ///
+        /// Because these URIs cannot be meaningfully represented by the `URI` event,
+        /// this function emits no events.
+        fn set_uri(ref self: ComponentState<TContractState>, uri: ByteArray) {
+            self.ERC1155_uri.write(uri);
+        }
     }
 
-    /// Checks if `to` either accepts the token either by implementing `IERC1155Receiver`
-    /// or if it's an account contract (supporting ISRC6). The transaction will fail if both are false.
+    /// Checks if `to` accepts the token by implementing `IERC1155Receiver`
+    /// or if it's an account contract (supporting ISRC6).
+    /// The transaction will fail if both are false.
     fn _check_on_ERC1155_received(
         from: ContractAddress, to: ContractAddress, token_id: u256, value: u256, data: Span<felt252>
-    ) {
-        let accepted = if (DualCaseSRC5 { contract_address: to }
+    ) -> bool {
```

Comment:
i guess the default value here is `false`?

Reply:
Not sure what you mean with default.

Reply:
Not sure if I understand correctly, but we are returning in every branch of this function, every return value is explicit, and there's no default usage.

Reply:
you're right, i misread